homeassistant/components/fibaro/binary_sensor.py

```diff
@@ -64,6 +72,40 @@ def __init__(self, fibaro_device: Any) -> None:
             self._attr_device_class = SENSOR_TYPES[stype][2]
             self._attr_icon = SENSOR_TYPES[stype][1]
 
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return the extra state attributes of the device."""
+        return super().extra_state_attributes | self._own_extra_state_attributes
+
     def update(self) -> None:
         """Get the latest data and update the state."""
-        self._attr_is_on = self.current_binary_state
+        if self.device_class == BinarySensorDeviceClass.MOVING:
```

Comment:
I'd check for `fibaro_device.type` since there could be another device in the future with a different data structure that we also want to map to the "moving" device class.

Side note: In the future I'd recommend refactoring the platform to use entity descriptions for the different devices.

Reply:
That makes sense, I changed it.

I'm happy to do the refactoring to entity descriptions after this PR is merged.